4/src/part1.c

```diff
@@ -0,0 +1,23 @@
+// Link: https://adventofcode.com/2022/day/4
+
+#include <stdio.h>
+
+int main() {
+    FILE *file = fopen("../data.txt", "r");
+    if (file == NULL) {
+        printf("Failed to open file\n");
+        return -1;
+    }
+
+    int start1, end1, start2, end2;
+    int count = 0;
+    while (fscanf(file, "%d-%d,%d-%d\n", &start1, &end1, &start2, &end2) != EOF) {
```

Comment:
Instead of checking for EOF, it's more robust to check if the number of successfully scanned items by fscanf is 4 (since you're expecting to scan 4 integers).